tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/TaskImpl.java

```diff
@@ -1027,7 +1027,7 @@ public void transition(TaskImpl task, TaskEvent event) {
       TaskAttempt earliestUnfinishedAttempt = null;
       for (TaskAttempt ta : task.attempts.values()) {
         // find the oldest running attempt
-        if (!ta.isFinished()) {
+        if (!ta.isFinished() && task.commitAttempt == null) {
```

Comment:
adding the logic here would suggest the related comment need to be updated as well, and we need an additional log (like 1034) to differentiate different cases.

I would suggest moving this if to before line 1046, and process it as a standalone special case (with log and everything)

Reply:
done